src/librustc/hir/map/def_collector.rs

```diff
@@ -295,4 +297,39 @@ impl<'a> visit::Visitor<'a> for DefCollector<'a> {
             }
         }
     }
+
+    fn visit_arg(&mut self, arg: &'a Arg) {
+        match arg.pat.node {
+            PatKind::Mac(..) => self.visit_macro_invoc(arg.id),
```

Comment:
Unfortunately, we cannot use macros in nested nodes to detect placeholders.

It breaks the next case
```rust
#[attribute] pattern!(): Type
```
when the nested node has an actual pattern and it's not a placeholder, but we need a placeholder for `#[attribute]`.

All of these nodes (`Arg`, `Arm`, `Variant` etc.) will have to keep an extra flag `is_placeholder: bool` in AST for now.